trin-validation/src/oracle.rs

```diff
@@ -51,18 +52,37 @@ impl HeaderOracle {
         let tx = self.history_jsonrpc_tx()?;
         tx.send(request)?;
 
-        let hwp_ssz = match resp_rx.recv().await {
+        let hwp = match resp_rx.recv().await {
             Some(val) => {
                 val.map_err(|err| anyhow!("Chain history subnetwork request error: {err:?}"))?
             }
             None => return Err(anyhow!("No response from chain history subnetwork")),
         };
-        let hwp_ssz = hwp_ssz
-            .as_str()
-            .ok_or_else(|| anyhow!("Invalid HWP format."))?;
-        let hwp_ssz = hex_decode(hwp_ssz)?;
-        HeaderWithProof::from_ssz_bytes(&hwp_ssz)
-            .map_err(|err| anyhow!("Invalid HWP received from chain history network: {err:?}"))
+        let hwp = match serde_json::from_value(hwp)? {
+            ContentInfo::Content { content, .. } => content,
+            ContentInfo::ConnectionId { .. } => {
+                return Err(anyhow!(
```

Comment:
I think you might be able to do roughly the same pattern as my comment before with "Nice cleanup".